src/hotspot/share/opto/regmask.hpp

```diff
@@ -26,13 +26,22 @@
 #define SHARE_OPTO_REGMASK_HPP
 
 #include "code/vmreg.hpp"
+#include "memory/arena.hpp"
 #include "opto/optoreg.hpp"
 #include "utilities/count_leading_zeros.hpp"
 #include "utilities/count_trailing_zeros.hpp"
 #include "utilities/globalDefinitions.hpp"
 
 class LRG;
 
+// To avoid unbounded RegMask growth and to be able to statically compute a
+// register mask size upper bound (see RM_SIZE_MAX below), we need to set some
+// form of limit on the number of stack slots used by BoxLockNodes. The limit
+// below is rather arbitrary but should be quite generous and cover all
+// practical cases. We reach this limit by, e.g., deeply nesting synchronized
+// statements in Java.
+const int BoxLockNode_SLOT_LIMIT = 200;
+
```

Comment:
Even before this constant, it would be nice to have an introductory comment, that lays out what the regmask is for, and what its basic design is.